internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -655,6 +655,27 @@ func getHyperthreading(clusterInstall *hiveext.AgentClusterInstall) *string {
 	}
 }
 
+func (r *ClusterDeploymentsReconciler) getEncodedCACert(ctx context.Context,
+	log logrus.FieldLogger,
+	clusterDeployment *hivev1.ClusterDeployment,
+	clusterInstall *hiveext.AgentClusterInstall) (*string, error) {
+	caCertificateRef := clusterInstall.Spec.IgnitionEndpoint.CaCertificateReference
+	if caCertificateRef != nil {
```

Comment:
So if `clusterInstall.Spec.IgnitionEndpoint.CaCertificateReference is nil` we return an error??
seems wrong

Reply:
So what should we do in this case, just skip it?

Reply:
Yes, if the `AgentClusterInstall` doesn't have `CaCertificateReference` there is no need to update the cluster

Reply:
I made this function(getEncodedCACert) return an error if ca can't be found/decoded and made the functions that use it not abort when error is returned.

Reply:
But there is a difference between the state where:
`clusterInstall.Spec.IgnitionEndpoint.CaCertificateReference is nil` - no Ca cert was configured
and the state where:
`clusterInstall.Spec.IgnitionEndpoint.CaCertificateReference is != nil` but this method fail to get the secret.
if it's the first option it's fine, but if it's the latter than it require an error.

Reply:
seems that in case it's nil you still return an error
https://github.com/openshift/assisted-service/pull/2972/files#r764009074

Reply:
This is a helper function and I don't propagate the error in functions that use it.

Reply:
@isaacdorfman please address 
https://github.com/openshift/assisted-service/pull/2972#discussion_r764994649

Reply:
Done